core/src/main/scala/org/bitcoins/core/p2p/ServiceIdentifier.scala

```diff
@@ -173,6 +173,19 @@ object ServiceIdentifier extends Factory[ServiceIdentifier] {
   def fromBytes(bytes: ByteVector): ServiceIdentifier =
     RawServiceIdentifierSerializer.read(bytes)
 
+  def fromString(string: String): ServiceIdentifier = string match {
+    case "NETWORK"         => NODE_NETWORK
+    case "NETWORK_LIMITED" => NODE_NETWORK_LIMITED
+    case "WITNESS"         => NODE_WITNESS
+    case "BLOOM"           => NODE_BLOOM
+    case "GETUTXO"         => NODE_GET_UTXO
+    case "COMPACT_FILTERS" => NODE_COMPACT_FILTERS
+    case "XTHIN"           => NODE_XTHIN
+    case _: String =>
+      throw new IllegalArgumentException(
+        "\"" + string + "\" does not represent a ServiceIdentifier")
```

Comment:
nit: Just fyi, in scala this can be `s"\"$string\" does not represent a ServiceIdentifier"`